### PR TITLE
Update Repo + Add Html Parser

### LINKED ISSUE #128: How can i parser html ?
i send 

`<html ></ html>`

with text/html  , but the rawbody is null

### PR DESCRIPTION
**Pull request by 3imed-jaberi**
- update the `.gitignore` file.
- avoid `package-lock.json` file.
- update the `README.md` file.
- update all dependencies.
- update the pipeline `.travis.yml`.
- clean the test code and src code  ( add esnext ).
- add lint.
- add basic git hooks check with huskyrc.
- fix #128 - html parser ( add code + add test ).
- re-structure `package.json`.

> can go to next version 🚀 !

### CODE REVIEW

**Comment by niftylettuce** [comment]
@jonathanong can I have access?

```sh
✖ You do not have write permissions required to publish this package.
```

It seems I don't have access to all packages in `koajs` organization.  I do however have access to scoped `@koa` but not non-scoped.

File: index.js [hunk 1]
```diff
@@ -25,62 +25,61 @@ var copy = require('copy-to');
...
-  var enableForm = checkEnable(enableTypes, 'form');
-  var enableJson = checkEnable(enableTypes, 'json');
-  var enableText = checkEnable(enableTypes, 'text');
-  var enableXml = checkEnable(enableTypes, 'xml');
+  const enableTypes = opts.enableTypes || ['json', 'form'];
+  const enableForm = checkEnable(enableTypes, 'form');
+  const enableJson = checkEnable(enableTypes, 'json');
+  const enableText = checkEnable(enableTypes, 'text');
+  const enableXml = checkEnable(enableTypes, 'xml');
+  const enableHtml = checkEnable(enableTypes, 'html');
```
Comment (dead-horse) [inline index.js:38]: hmmm, I think there is no need to add a `html` type here. we can get text/html by:

```
bodyparser({
  extendTypes: {
    text: [ 'text/plain', 'text/html' ],
  }
})
```

Comment (dead-horse) [reply index.js:38]: @3imed-jaberi can you revert this `enableHtml` feature here? :)

**Comment by dead-horse** [comment]
others LGTM

File: index.js (hunk 1 above)
Comment (dead-horse) [reply index.js:38]: I'v reverted `enableHtml` option, we can use text parser to parse html content.:)

https://github.com/koajs/bodyparser/commit/4d931c634e9b59a843152f56d68b3ef2e1719675#diff-faf36523e15f6c0b9ee07674a066b3ecR262

Comment (3imed-jaberi) [reply index.js:38]: @dead-horse, 
sorry bro, I did not see your comment 😟 ..

**Pull request merged**